jerry-core/jerry.c

```diff
@@ -2023,42 +2019,19 @@ jerry_parse_and_save_snapshot (const jerry_api_char_t *source_p, /**< script sou
   jsp_status_t parse_status;
   ecma_compiled_code_t *bytecode_data_p;
 
-  snapshot_buffer_write_offset = 0;
-  snapshot_last_compiled_code_offset = 0;
+  snapshot_buffer_write_offset = JERRY_ALIGNUP (sizeof (jerry_snapshot_header_t),
+                                                MEM_ALIGNMENT);
   snapshot_error_occured = false;
   snapshot_buffer_p = buffer_p;
   snapshot_buffer_size = buffer_size;
-  snapshot_map_entries_p = NULL;
-
-  uint64_t version = JERRY_SNAPSHOT_VERSION;
-  if (!jrt_write_to_buffer_by_offset (buffer_p,
-                                      buffer_size,
-                                      &snapshot_buffer_write_offset,
-                                      &version,
-                                      sizeof (version)))
-  {
-    return 0;
-  }
-
-  snapshot_buffer_write_offset = JERRY_ALIGNUP (snapshot_buffer_write_offset, MEM_ALIGNMENT);
-
-  size_t header_offset = snapshot_buffer_write_offset;
-
-  snapshot_buffer_write_offset += JERRY_ALIGNUP (sizeof (jerry_snapshot_header_t), MEM_ALIGNMENT);
-
-  if (snapshot_buffer_write_offset > buffer_size)
-  {
-    return 0;
-  }
-
-  size_t compiled_code_start = snapshot_buffer_write_offset;
-
-  snapshot_report_byte_code_compilation = true;
   jerry_api_object_t *error_obj_p = NULL;
 
   if (is_for_global)
   {
-    parse_status = parser_parse_script (source_p, source_size, &bytecode_data_p, &error_obj_p);
+    parse_status = parser_parse_script (source_p,
+                                        source_size,
+                                        &bytecode_data_p,
+                                        &error_obj_p);
```

Comment:
_This_ style takes too much vertical space from my screen for a single statement but _de gustibus non est disputandum_.